keadm/app/cmd/util/ubuntuinstaller.go

```diff
@@ -433,22 +435,68 @@ func (u *UbuntuOS) InstallKubeEdge() error {
 	//Currently it is missing and once checksum is in place, checksum check required
 	//to be added here.
 	filename := fmt.Sprintf("kubeedge-v%s-linux-%s.tar.gz", u.KubeEdgeVersion, arch)
+	checksumFilename := fmt.Sprintf("checksum_kubeedge-v%s-linux-%s.txt", u.KubeEdgeVersion, arch)
 	filePath := fmt.Sprintf("%s%s", KubeEdgePath, filename)
 	fileStat, err := os.Stat(filePath)
 	if err == nil && fileStat.Name() != "" {
 		fmt.Println("Expected or Default KubeEdge version", u.KubeEdgeVersion, "is already downloaded")
 		goto SKIPDOWNLOADAND
 	}
 
-	//Download the tar from repo
-	dwnldURL = fmt.Sprintf("cd %s && wget -k --no-check-certificate --progress=bar:force %s/v%s/%s", KubeEdgePath, KubeEdgeDownloadURL, u.KubeEdgeVersion, filename)
-	cmd = &Command{Cmd: exec.Command("sh", "-c", dwnldURL)}
-	err = cmd.ExecuteCmdShowOutput()
-	errout = cmd.GetStdErr()
-	if err != nil || errout != "" {
-		return fmt.Errorf("%s", errout)
+	for i := 0; i < downloadRetryTimes; i++ {
+		//Download the tar from repo
+		dwnldURL = fmt.Sprintf("cd %s && wget -k --no-check-certificate --progress=bar:force %s/v%s/%s", KubeEdgePath, KubeEdgeDownloadURL, u.KubeEdgeVersion, filename)
+		cmd = &Command{Cmd: exec.Command("sh", "-c", dwnldURL)}
+		err = cmd.ExecuteCmdShowOutput()
+		errout = cmd.GetStdErr()
+		if err != nil || errout != "" {
+			return fmt.Errorf("%s", errout)
+		}
+		fmt.Println()
+
+		//Download the checksum file from repo
+		dwnldURL = fmt.Sprintf("cd %s && wget -k --no-check-certificate --progress=bar:force %s/v%s/%s", KubeEdgePath, KubeEdgeDownloadURL, u.KubeEdgeVersion, checksumFilename)
```

Comment:
Instead of downloading the checksum file and the doing a cat of it. It is better to use it the below ways 
1. 
$ 
$ wget -qO- https://github.com/kubeedge/kubeedge/releases/download/v0.3.0-beta.0/checksum_keadm-v0.3.0-beta.0-linux-amd64.txt
5775c5c1b25a9148fd592cf61e0cd671ca8fb345c63c897709bda2c479bb3a315c9f979e934fbd5a79dfcb633f793ffe948979773e65f95c65791b4ac08c3d0f
$ 
2. 
$ 
$ result=$(wget -qO- https://github.com/kubeedge/kubeedge/releases/download/v0.3.0-beta.0/checksum_keadm-v0.3.0-beta.0-linux-amd64.txt)
$ echo $result
5775c5c1b25a9148fd592cf61e0cd671ca8fb345c63c897709bda2c479bb3a315c9f979e934fbd5a79dfcb633f793ffe948979773e65f95c65791b4ac08c3d0f
$ 

I suggest to use first option

Reply:
@samy2019 Thanks for your comment. The code has been updated accordingly.